File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        vector<int> LIS(nums.size(), 1);

        for (unsigned int i = 0; i < nums.size(); ++i ) {
            int maxSub = 0;
            for (unsigned int j = 0; j < i; ++j) {
                if (nums[j] < nums[i]) {
                    maxSub = max(maxSub, LIS[j] + 1);
                }
            }
            LIS[i] = max(maxSub, LIS[i]);
        }

        int res = 0;
        for (const int& val : LIS) {
            res = max(res, val);
        }

        return res;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-1.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = res = nums[0]
        for i in range(1, len(nums)):
            if curr < 0:
                curr = nums[i]
            else:
                curr += nums[i]
            res = max(res, curr)
        return res

